pandas/tests/scalar/timedelta/test_construction.py

```diff
@@ -210,3 +212,20 @@ def test_td_constructor_on_nanoseconds(constructed_td, conversion):
 def test_td_constructor_value_error():
     with pytest.raises(TypeError):
         Timedelta(nanoseconds='abc')
+
+
+@pytest.mark.parametrize("str_unit, unit, expectation", [
+    ("", "s", tm.do_not_raise),              # Expected case
+    ("s", "d", pytest.raises(ValueError)),   # Units doubly defined
```

Comment:
test the error message as well

Reply:
okay, will use `pytest.raises(Exc, message="Exception message")` though, if it is okay.